### PR TITLE
add easyexif/1.0

### LINKED ISSUE #2666: [request] easyexif
### Package Details
  * Source code: **https://github.com/mayanklahiri/easyexif**

### Description Of The Library / Tool
Tiny ISO-compliant C++ EXIF parsing library

### PR DESCRIPTION
**Pull request by theirix**
Specify library name and version:  **easyexif/1.0**

Closes #2666

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/easyexif/all/test_package/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        cmake = CMake(self)
+        # Current dir is "test_package/build/<build_id>" and CMakeLists.txt is
+        # in "test_package"
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
+            with tools.chdir("bin"):
+                self.run(".%sexample" % os.sep, run_environment=True)
```
Comment (madebr) [inline recipes/easyexif/all/test_package/conanfile.py:19]:
```suggestion
            bin_path = os.path.join("bin", "example")
            self.run(bin_path, run_environment=True)
```

**Review by madebr — DISMISSED** [review]

File: recipes/easyexif/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import os
+from conans import CMake, ConanFile, tools
+
+class EasyExifConan(ConanFile):
+    name = "easyexif"
+    description = "Tiny ISO-compliant C++ EXIF parsing library, third-party dependency free."
+    topics = ("conan", "exif", "image", "multimedia", "format", "graphics")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/mayanklahiri/easyexif"
+    license = "https://github.com/mayanklahiri/easyexif/blob/master/LICENSE"
```
Comment (madebr) [inline recipes/easyexif/all/conanfile.py:10]: Isn't it `MIT`?
https://mit-license.org/

Comment (theirix) [reply recipes/easyexif/all/conanfile.py:10]: I tried to match it against MIT or BSD, and it slightly differs. Do not know if it was intended by the author but I think it is better to set an explicit path to license.

Comment (uilianries) [reply recipes/easyexif/all/conanfile.py:10]: It's BSD-2-Clause https://spdx.org/licenses/BSD-2-Clause.html

**Review by madebr — DISMISSED** [review]

**Review by uilianries — DISMISSED** [review]

File: recipes/easyexif/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    license = "BSD-2-Clause"
+    exports_sources = "CMakeLists.txt"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {"shared": False, "fPIC": True}
+    generators = "cmake"
+
```
Comment (SpaceIm) [inline recipes/easyexif/all/conanfile.py:19]:
```suggestion
    generators = "cmake"
    
    _cmake = None

```

File: recipes/easyexif/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.configure()
+        return cmake
```
Comment (SpaceIm) [inline recipes/easyexif/all/conanfile.py:36]:
```suggestion
    def _configure_cmake(self):
        if self._cmake:
            return self._cmake
        self._cmake = CMake(self)
        self._cmake.configure()
        return self._cmake
```

File: recipes/easyexif/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    generators = "cmake"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
```
Comment (SpaceIm) [inline recipes/easyexif/all/conanfile.py:27]:
```suggestion
    def config_options(self):
        if self.settings.os == "Windows":
            del self.options.fPIC

    def configure(self):
        if self.options.shared:
            del self.options.fPIC
        if self.settings.compiler.cppstd:
            tools.check_min_cppstd(self, 11)

```

File: recipes/easyexif/all/test_package/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,12 @@
+cmake_minimum_required(VERSION 3.3)
```
Comment (SpaceIm) [inline recipes/easyexif/all/test_package/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 3.1)
```

File: recipes/easyexif/all/test_package/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+project(PackageTest CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+set(CMAKE_CXX_STANDARD_REQUIRED on)
+
```
Comment (SpaceIm) [inline recipes/easyexif/all/test_package/CMakeLists.txt:8]:
```suggestion
```

File: recipes/easyexif/all/test_package/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+add_executable(example example.cpp)
+
+set_target_properties(example PROPERTIES CXX_STANDARD 11)
```
Comment (SpaceIm) [inline recipes/easyexif/all/test_package/CMakeLists.txt:11]:
```suggestion
set_target_properties(example PROPERTIES CXX_STANDARD 11 CXX_STANDARD_REQUIRED ON)
```

File: recipes/easyexif/all/test_package/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+
+class SolaceTestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    def build(self):
+        cmake = CMake(self)
+        # Current dir is "test_package/build/<build_id>" and CMakeLists.txt is
+        # in "test_package"
```
Comment (SpaceIm) [inline recipes/easyexif/all/test_package/conanfile.py:12]:
```suggestion
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

**Review by uilianries — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Pull request merged**